[OliviaJHL].iP

OliviaJHL opened the pull request:
### DUKE
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
-  ~FAST~ _SUPER_ FAST to use 🥳

All you need to do is,
1. download it from [here](https://github.com/OliviaJHL/ip/releases/tag/A-Jar).
2. run in the terminal!
3. let it manage your tasks for you 😉
4. And it is **FREE**!

Features:
- [X] Deleting tasks
- [X] Managing saving 
- [X] Managing finding tasks
- [X] Managing todo
- [X] Managing deadlines 
- [X] Managing event
- [X] Managing mark as done
- [X] Managing mark as undone

If you Java programmer, you can use it to practice Java too. Here's the main method:

``` ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

OliviaJHL committed:
level-1 Greet, Echo, Exit

OliviaJHL committed:
level-2 Add and list

OliviaJHL committed:
level-3 Mark as Done

OliviaJHL committed:
level-4 oDos, Events, Deadlines

OliviaJHL committed:
A-TextUiTesting

OliviaJHL committed:
A-TextUiTesting

OliviaJHL committed:
level5.Handle Errors

OliviaJHL committed:
level-6.Delete

OliviaJHL committed:
A-TextUiTesting modifying

OliviaJHL committed:
Level-7.save

OliviaJHL committed:
Level-7.save

OliviaJHL committed:
Level-7.save(modifying)

OliviaJHL committed:
Merge branch 'branch-Level-7'

OliviaJHL committed:
Level-8. Dates and Times

OliviaJHL committed:
A-MoreOOP.Use More OOP

OliviaJHL committed:
A-Packages: Organize into Packages

OliviaJHL committed:
Level 8. Dates and Times(modify)

OliviaJHL committed:
Level 8. Dates and Time

OliviaJHL committed:
Level 8. Dates and Time(changing)

OliviaJHL committed:
Merge branch 'branch-Level-8'

OliviaJHL committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

OliviaJHL committed:
A-JUnit: Add JUnit Tests

OliviaJHL committed:
Add JavaDoc comments

OliviaJHL committed:
Add JavaDoc comments(modify)

OliviaJHL committed:
Add JavaDoc comments(modifing)

OliviaJHL committed:
Add JavaDoc comments(modifing2)

OliviaJHL committed:
Add JavaDoc comments(modifing2)

OliviaJHL committed:
Merge branch 'branch-A-JavaDoc'

OliviaJHL committed:
modifying

OliviaJHL committed:
Tweak the code to comply with a coding standard

OliviaJHL committed:
Tweak the code to comply with a coding standard(modify)

OliviaJHL committed:
Tweak the code to comply with a coding standard(modify)

OliviaJHL committed:
Tweak the code to comply with a coding standard(modify1)

OliviaJHL committed:
Merge branch 'branch-A-CodingStandard'

OliviaJHL committed:
Level 9. Find

OliviaJHL committed:
Level 9. Find

OliviaJHL committed:
Level 9. Find(modify)

OliviaJHL committed:
Level 9. Find(modify)

OliviaJHL committed:
Merge branch 'branch-Level-9'

OliviaJHL committed:
A-Gradle

OliviaJHL committed:
A-CheckStyle: Use CheckStyle

Infrix commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+ */
+public class Parser {
+    private static final String Indentation = " ";
+    private static final String Horizontal = "____________________________________________________________";
```
constant names should be in all uppercase

Infrix commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+     * @param  tasks
+     * @return command
+     */
+    public static Task parse(String cmd, TaskList tasks) {
```
Perhaps you would like to create an abstract Command class for different commands to inherit from it? Then the parser could just be used to create commands, instead of executing the commands in the parser

Infrix reviewed (commented):
Overall looks good, coding standard wise I could only catch one error

OliviaJHL committed:
Level 10.GUI

OliviaJHL committed:
Level 10.GUI

OliviaJHL committed:
Level 10.GUI

OliviaJHL committed:
Merge branch 'branch-Level-10'

jugheadjones10 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,186 @@
+ * Represents all methods such as showLogo, showWelcome...
+ */
+public class Ui {
+    private static final String Indentation = " ";
```
Constants should be in all uppercase if I'm not wrong

jugheadjones10 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @param t
+     * @throws IOException
+     */
+    public void updateFile(TaskList t) throws IOException {
```
Single letter argument name (t) might make sense here because the method is short, but maybe we can guard against future confusion by just using "taskList" as the argument name instead since in the future this method might get longer.

jugheadjones10 reviewed (commented):
Looks good to merge! Some minor nits.

mandykqh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+                        LocalDateTime datetime1 = LocalDateTime.parse(fromtime, formatter1);
+                        LocalDateTime datetime2 = LocalDateTime.parse(totime, formatter2);
+                        //System.out.println(datetime1.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")));
+                        //System.out.println(datetime2.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")));
```
You may want to delete dead code as you can recover them from your repo if needed.

mandykqh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+                    System.out.println("  ☹ OOPS!!! The index number cannot be empty.");
+                    System.out.println(Indentation + Horizontal);
+
+                    str += "  ☹ OOPS!!! The index number cannot be empty. \n";
```
To avoid having to repeat your error messages, you can possibly throw say an `InvalidInputException` with your intended error message, and then handle it in the Duke class. The section on [A-MoreOOP](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/projectDuke/index.html#a-moreoop) provides an example for my explanation.

mandykqh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,209 @@
+                        str += "/n";
+                    }
+                    tasks.add(task);
+                } catch (Exception e) {
```
It appears that you have many try-catch blocks here. You may consider just using a single try-catch block and then throw the necessary exceptions (eg. `InvalidInputException`, `EmptyInputException`)  that will subsequently be handled in `Duke` as mentioned in my previous comment.

mandykqh reviewed (commented):
Nice job overall! Just a few suggestions regarding error handling to reduce code duplication. 👍

OliviaJHL committed:
Add Assertions for some parts
Add assertion statement to check the input task whether is null.
Add assertion statement to check the input index for remove and get methods which are in TaskList class
Add assertion statement to check when the task is removed, the current number should be updated and -1 from original number

OliviaJHL committed:
Merge pull request #1 from OliviaJHL/branch-A-Assertions

Add Assertions for some parts

OliviaJHL committed:
Improve code quality for some parts
Solve long method problem is Parser.java file
Delete unnecessary codes in many files
Change some Java Doc
Add style for GUI

OliviaJHL committed:
Merge pull request #2 from OliviaJHL/branch-A-CodeQuality

Improve code quality for some parts

OliviaJHL committed:
Add Extension
Add sort method, based on the task type to sort the task list.
Planing to add more methods

OliviaJHL committed:
Merge pull request #3 from OliviaJHL/branch-A-BCD-Extension

Add Extension

OliviaJHL committed:
Modify code quality
Change three parts of Java doc in storage and duke file
The issues should be resolved

OliviaJHL committed:
Setup CI
Add .yml file into .github/workflows
Modify the mainwindow in sceneBuilder

OliviaJHL committed:
improve code quality

OliviaJHL committed:
Update TaskTest.java

OliviaJHL committed:
checkstyle change

OliviaJHL committed:
BIG change
-perfect code quality
-GUI, resizing by windows
-handles more errors
-and some details

OliviaJHL committed:
Small change

OliviaJHL committed:
make some change for the Readme and Ui.png

OliviaJHL committed:
Update README.md

OliviaJHL committed:
Update README.md

OliviaJHL committed:
Update README.md

OliviaJHL committed:
Update README.md
